### ISSUE TITLE
ValueDivMulti<T> divide by 0 will crash the session

### ISSUE THREAD

**Issue by r3coder** [Yellow-Dog-Man/Resonite-Issues#1425]
### Describe the bug?

Division by 0 on the ValueDivMulti<T> crashes the session.

### To Reproduce

In any world, spawn a `Operator -> ValueDivMulti<T>` ->Spawn any MultiDiv Node.

Set the divisor to 0, and pop out the result will crash anybody in the session.

In my case, it was an `integer` type.

https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/10560541/6e13614e-a155-4b33-b6a2-c2fca429ea28

### Expected behavior

It should handle the division by 0 error, and result should be 0 or something,

### Screenshots

![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/10560541/b9b232b0-53dd-41b7-8549-70f5e31bc54f)

### Resonite Version Number

2024.2.12.1430

### What Platforms does this occur on?

Windows

### Log Files

```
PM 12:50:00.990 (140 FPS)	Implicit 3255
-ExecutionRuntime<FrooxEngineContext> (16 nodes)
--[0] ObjectFieldDrive<String>
---Inputs:
----Value: * on [12] FormatString (Type: System.String - Object)
--[1] ValueDivMulti<Int32>
---Dynamic Inputs:
----Inputs (Count: 2)
-----0: * on [11] ValueAddMulti<Int32> (Type: System.Int32 - Value)
-----1: Value on [7] ReadDynamicValueVariable<Int32> (Type: System.Int32 - Value)
--[2] ReadDynamicValueVariable<Int32>
---Inputs:
----Source: * on [3] ElementSource<Slot> (Type: FrooxEngine.Slot - Object)
----Path: * on [4] ExternalObjectInput<FrooxEngineContext,String> (Type: System.String - Object)
--[3] ElementSource<Slot>
--[4] ExternalObjectInput<FrooxEngineContext,String>
--[5] ReadDynamicValueVariable<Int32>
---Inputs:
----Source: * on [3] ElementSource<Slot> (Type: FrooxEngine.Slot - Object)
----Path: * on [8] ExternalObjectInput<FrooxEngineContext,String> (Type: System.String - Object)
--[6] ReadDynamicValueVariable<Int32>
---Inputs:
----Source: * on [3] ElementSource<Slot> (Type: FrooxEngine.Slot - Object)
----Path: * on [9] ExternalObjectInput<FrooxEngineContext,String> (Type: System.String - Object)
--[7] ReadDynamicValueVariable<Int32>
---Inputs:
----Source: * on [3] ElementSource<Slot> (Type: FrooxEngine.Slot - Object)
----Path: * on [10] ExternalObjectInput<FrooxEngineContext,String> (Type: System.String - Object)
--[8] ExternalObjectInput<FrooxEngineContext,String>
--[9] ExternalObjectInput<FrooxEngineContext,String>
--[10] ExternalObjectInput<FrooxEngineContext,String>
--[11] ValueAddMulti<Int32>
---Dynamic Inputs:
----Inputs (Count: 3)
-----0: Value on [2] ReadDynamicValueVariable<Int32> (Type: System.Int32 - Value)
-----1: Value on [5] ReadDynamicValueVariable<Int32> (Type: System.Int32 - Value)
-----2: Value on [6] ReadDynamicValueVariable<Int32> (Type: System.Int32 - Value)
--[12] FormatString
---Inputs:
----Format: * on [14] ExternalObjectInput<FrooxEngineContext,String> (Type: System.String - Object)
---Dynamic Inputs:
----Parameters (Count: 2)
-----0: * on [13] ValueToObjectCast<Int32> (Type: System.Object - Object)
-----1: (none)
--[13] ValueToObjectCast<Int32>
---Inputs:
----Input: * on [1] ValueDivMulti<Int32> (Type: System.Int32 - Value)
--[14] ExternalObjectInput<FrooxEngineContext,String>
--[15] ValueMulMulti<Int32>
---Dynamic Inputs:
----Inputs (Count: 2)
-----0: * on [11] ValueAddMulti<Int32> (Type: System.Int32 - Value)
-----1: Value on [7] ReadDynamicValueVariable<Int32> (Type: System.Int32 - Value)

PM 12:50:00.995 (140 FPS)	Unhandled Exception when updating world: Tómleiki og Ljós 공허와 빛 Resonite. State Running, Refresh Stage: ProtoFluxContinuousChanges, Init State: Finished, SyncTick 186757, StateVersion: 333011
Dumped world state to: 
System.DivideByZeroException: Attempted to divide by zero.
  at Elements.Core.Coder`1+<>c[T].<.cctor>b__141_242 (System.Int32 a, System.Int32 b) [0x00000] in <aa0c58f0f73841a7b5fdcd3aa64d3410>:0 
  at ProtoFlux.Runtimes.Execution.Nodes.Operators.ValueDivMulti`1[T].Compute (ProtoFlux.Runtimes.Execution.ExecutionContext context) [0x0003d] in <d050fb33c946416da37c60d8dc5b7fa6>:0 
  at ProtoFlux.Runtimes.Execution.ValueFunctionNode`2[C,T].Evaluate (C context) [0x00000] in <73d8ff99d53044acbb81bdc4e997e873>:0 
  at ProtoFlux.Runtimes.Execution.EvaluationSequence`1[C].Evaluate (C context) [0x00068] in <73d8ff99d53044acbb81bdc4e997e873>:0 
  at ProtoFlux.Runtimes.Execution.EvaluationSequence`1[C].EvaluateObject[T] (ProtoFlux.Core.IOutput output, C context) [0x00000] in <73d8ff99d53044acbb81bdc4e997e873>:0 
  at ProtoFlux.Runtimes.Execution.ExecutionRuntime`1[C].EvaluateObject[T] (ProtoFlux.Core.IOutput output, ProtoFlux.Runtimes.Execution.ExecutionContext context) [0x0004c] in <73d8ff99d53044acbb81bdc4e997e873>:0 
  at ProtoFlux.Runtimes.Execution.ExecutionContextExtensions.Evaluate[T] (ProtoFlux.Core.ObjectInput`1[T] input, ProtoFlux.Runtimes.Execution.ExecutionContext context, T defaultValue) [0x00018] in <73d8ff99d53044acbb81bdc4e997e873>:0 
  at FrooxEngine.ProtoFlux.CoreNodes.ObjectFieldDrive`1[T].EvaluateValue (FrooxEngine.ProtoFlux.FrooxEngineContext context) [0x00000] in <6450ba452eef4b15a58558870451eee7>:0 
  at FrooxEngine.ProtoFlux.CoreNodes.FieldDriveBase`1[T].Changed (FrooxEngine.ProtoFlux.FrooxEngineContext context) [0x0003b] in <6450ba452eef4b15a58558870451eee7>:0 
  at ProtoFlux.Runtimes.Execution.ExecutionChangesDispatcher`1[C].DispatchChangeToAllRecursively (ProtoFlux.Runtimes.Execution.ExecutionRuntime`1[C] runtime, C context) [0x0001f] in <73d8ff99d53044acbb81bdc4e997e873>:0 
  at ProtoFlux.Runtimes.Execution.ExecutionChangesDispatcher`1[C].DispatchChangeToAllListeners (ProtoFlux.Runtimes.Execution.ExecutionRuntime`1[C] runtime, C context) [0x00048] in <73d8ff99d53044acbb81bdc4e997e873>:0 
  at FrooxEngine.ProtoFlux.ProtoFluxNodeGroup.RunNodeChanges () [0x00108] in <6450ba452eef4b15a58558870451eee7>:0 
  at FrooxEngine.ProtoFlux.ProtoFluxController.RunContinuousChanges () [0x0002d] in <6450ba452eef4b15a58558870451eee7>:0 
  at FrooxEngine.World.RefreshStep () [0x00421] in <6450ba452eef4b15a58558870451eee7>:0 
  at FrooxEngine.World.Refresh () [0x00073] in <6450ba452eef4b15a58558870451eee7>:0 
  at FrooxEngine.WorldManager.UpdateStep () [0x004a3] in <6450ba452eef4b15a58558870451eee7>:0 

  at System.Environment.get_StackTrace () [0x00000] in <9577ac7a62ef43179789031239ba8798>:0 
  at Elements.Core.UniLog.Error (System.String message, System.Boolean stackTrace) [0x00000] in <aa0c58f0f73841a7b5fdcd3aa64d3410>:0 
  at FrooxEngine.WorldManager.UpdateStep () [0x00000] in <6450ba452eef4b15a58558870451eee7>:0 
  at FrooxEngine.WorldManager.RunUpdateLoop () [0x00000] in <6450ba452eef4b15a58558870451eee7>:0 
  at FrooxEngine.Engine.UpdateStep () [0x00000] in <6450ba452eef4b15a58558870451eee7>:0 
  at FrooxEngine.Engine.RunUpdateLoop () [0x00000] in <6450ba452eef4b15a58558870451eee7>:0 
  at UnityFrooxEngineRunner.FrooxEngineRunner.UpdateFrooxEngine () [0x00000] in <e74db962a71e4b62b259fa91018e0f2f>:0 
  at UnityFrooxEngineRunner.FrooxEngineRunner.Update () [0x00000] in <e74db962a71e4b62b259fa91018e0f2f>:0 
PM 12:50:00.996 (140 FPS)	Exception calling WorldDestroyed Event:
System.InvalidOperationException: An attempt was made to transition a task to a final state when it had already completed.
  at System.Threading.Tasks.TaskCompletionSource`1[TResult].SetResult (TResult result) [0x00013] in <9577ac7a62ef43179789031239ba8798>:0 
  at FrooxEngine.EngineGatherJob+<>c__DisplayClass13_0.<RunSessionDownload>b__1 (FrooxEngine.World w) [0x00013] in <6450ba452eef4b15a58558870451eee7>:0 
  at (wrapper delegate-invoke) System.Action`1[FrooxEngine.World].invoke_void_T(FrooxEngine.World)
  at FrooxEngine.World.Destroy () [0x00037] in <6450ba452eef4b15a58558870451eee7>:0 

  at System.Environment.get_StackTrace () [0x00000] in <9577ac7a62ef43179789031239ba8798>:0 
  at Elements.Core.UniLog.Error (System.String message, System.Boolean stackTrace) [0x00000] in <aa0c58f0f73841a7b5fdcd3aa64d3410>:0 
  at FrooxEngine.World.Destroy () [0x00000] in <6450ba452eef4b15a58558870451eee7>:0 
  at FrooxEngine.WorldManager.UpdateStep () [0x00000] in <6450ba452eef4b15a58558870451eee7>:0 
  at FrooxEngine.WorldManager.RunUpdateLoop () [0x00000] in <6450ba452eef4b15a58558870451eee7>:0 
  at FrooxEngine.Engine.UpdateStep () [0x00000] in <6450ba452eef4b15a58558870451eee7>:0 
  at FrooxEngine.Engine.RunUpdateLoop () [0x00000] in <6450ba452eef4b15a58558870451eee7>:0 
  at UnityFrooxEngineRunner.FrooxEngineRunner.UpdateFrooxEngine () [0x00000] in <e74db962a71e4b62b259fa91018e0f2f>:0 
  at UnityFrooxEngineRunner.FrooxEngineRunner.Update () [0x00000] in <e74db962a71e4b62b259fa91018e0f2f>:0 
PM 12:50:01.001 (140 FPS)	Session updated, forcing status update
```

### Additional Context

In some cases (if the calculation didn't happen), an item that may crash the session can be saved to an item.
So, even if somebody doesn't have to spawn flux nodes, they can also crash the world by item spawn.
I think it's urgent to fix the issue since it can be easily abused.

### Reporters

R3C0D3r, MirPASEC

**Comment by ProbablePrime** [comment]
Ill tack this onto my other maths pr.

**Comment by Frooxius** [comment]
Fix has been merged in 2024.3.12.87. Thanks for report!
